webapp/src/actions/index.js

```diff
@@ -2,14 +2,15 @@
 // See License for license information.
 
 import {PostTypes} from 'mattermost-redux/action_types';
+import {getConfig} from 'mattermost-redux/selectors/entities/general';
 
 import Client from '../client';
 
 export function startMeeting(channelId, force = false) {
     return async (dispatch, getState) => {
         try {
             const startFunction = force ? Client.forceStartMeeting : Client.startMeeting;
-            const meetingURL = await startFunction(channelId, true);
+            const meetingURL = await startFunction(channelId, getServerRoute(getState()), true);
```

Comment:
This seems painful: having to pass in the base URL in all cases.

In the Redux client, there is a `setUrl` method that initializes the client once so it can be used without having to resolve this all the time. Is that an option here?

Reply:
I agree that this seems a bit cumbersome. I think the problem is that we instantiate the plugin's `Client` before we have access to the redux store. We also aren't using the `mattermost-redux` `Client4` directly, we only use the `getOptions` method it provides. So the `setUrl` method would not have any effect.

The plugin's `Client` does not have any state itself. So one solution would be to instantiate a new `Client` with the redux state each time we need to make a request. The `Client`'s constructor could then use the state to assign its `url` field which is currently hardcoded to `'/plugins/' + id`.

https://github.com/larkox/mattermost-plugin-zoom/blob/115838ed4f5608eec07807dc07fbb14a302fc1f6/webapp/src/client/client.js#L11

@lieut-data In your comment on the [Jira ticket](https://mattermost.atlassian.net/browse/MM-23411), you propose moving this `getPluginServerRoute` to `mattermost-redux` then importing it elsewhere. How do you feel about that approach in regards to this PR? I'm 0/5 on introducing the function here as we are doing in this PR.

Reply:
Moving the function to redux /feels/ like it won't solve the root problem of having to resolve the state. I was imagining something like adding a `setUrl` method on the `Client` instance and calling it from the `Plugin.initialize` method with the passed-in `store`. Thoughts?

Reply:
My fear is what happens if SiteURL changes in runtime (I go to SystemSettings and update the Site URL). Until the plugin is reloaded everything will be broken.

We could setup webhooks so the plugin can tell the JS Client to update its SiteURL on configuration change, but feels a bit of an overkill (0/5).

Reply:
We actually don't currently support changing the SiteURL's /subpath/ in real-time -- see https://mattermost.atlassian.net/browse/MM-14194 for the nitty-gritty, but basically there's an implicit assumption that the server starts up with a fixed subpath that won't take effect until a restart. 

You can rely on that here as well, and since we're using a relative URL anyway, it doesn't matter if the domain changes.

Reply:
The client is updated with the latest config changes through websockets, so if the `SiteURL` changes, then the client's redux store picks it up.

1/5 Regardless if a subpath is involved, I think we should do our best to use this functionality since this is what it's for: make the clients have up-to-date state in regards to config. It's "just as good" as refreshing the page in this case, which is a great feat of a SPA.

I think it's good to mirror the behavior of a refresh, though I can see arguments against it. For example, the subpath changing would indeed cause problems, but I don't think we want stale browser sessions to be the necessary evil to have the requests continue to work.

Reply:
That's a fair point -- ideally the endusers shouldn't /have/ to refresh.

> 1/5 Regardless if a subpath is involved, I think we should do our best to use this functionality since this is what it's for: make the clients have up-to-date state in regards to config.

In this particular case, it only matters if a subpath is involved -- since we don't care about the domain, and can always just anchor as a relative URL.

But my thinking is that since we expect changing a subpath to be a disruptive, server-restarting, weird-side-effects-included event, I don't think we should over-complicate this code to handle an event that's already known to be painful.

Reply:
@lieut-data What about this (not so rare in my opinion) usecase?
I move my server to a subdirectory. Everything breaks down and I have to restart the server. The server restart, all the plugins load, everything is fine, but I then realize I have not yet updated the SiteURL. I change that, but the plugin is still not working. No idea why.
Shouldn't we take in mind that? Or do you think this usecase is not something that may happen?

Reply:
@larkox, as I understand it, until you change the SiteURL to reflect the subpath, the server definitely won't work properly. The webapp will be broken in lots of weird ways. Only after changing the SiteURL /and/ restarting the server (which necessarily restarts the plugins) will the subpath take effect.

Reply:
> Only after changing the SiteURL /and/ restarting the server (which necessarily restarts the plugins) will the subpath take effect.

Especially since the plugins restart in this case, I agree that it is okay to initialize the SiteURL on plugin startup, and continue using that value during the plugin's runtime. It makes the `Client` a bit more clean and useful to its callers.